### PR TITLE
Add permission checking middle-ware

### PR DESCRIPTION
**Pull request by ovekyc**

### CODE REVIEW

File: src/server/permission.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import assert from 'assert';
+
+export const PERMISSION = {
+  RW: 'rw',
+  R: 'r-',
+  W: '-w',
+  None: '--'
+};
+
+export const requiredPermmission = (permission) => {
```
Comment (hewonjeong) [inline src/server/permission.js]: typo : requiredPermission

**Comment by festinalente91** [comment]
- added user type in access / refresh token
- renamed access / refresh token claim (user => userType, type => tokenType)
- added permission in authenticate method at auth-token.js, each user type has own permission

**Comment by festinalente91** [comment]
lgtm

**Comment by hewonjeong** [comment]
lgtm

**Pull request merged**